packages/vm/src/index.ts

```diff
@@ -205,30 +205,34 @@ export default class VM extends AsyncEventEmitter {
       this._common = opts.common
     } else {
       const DEFAULT_CHAIN = Chain.Mainnet
-      const supportedHardforks = [
-        'chainstart',
-        'homestead',
-        'dao',
-        'tangerineWhistle',
-        'spuriousDragon',
-        'byzantium',
-        'constantinople',
-        'petersburg',
-        'istanbul',
-        'muirGlacier',
-        'berlin',
-        'arrowGlacier',
-      ]
-
-      this._common = new Common({
-        chain: DEFAULT_CHAIN,
-        supportedHardforks,
-      })
+
+      this._common = new Common({ chain: DEFAULT_CHAIN })
     }
     this._common.on('hardforkChanged', () => {
       this._opcodes = getOpcodesForHF(this._common)
     })
 
+    const supportedHardforks = [
+      Hardfork.Chainstart,
+      Hardfork.Homestead,
+      Hardfork.Dao,
+      Hardfork.TangerineWhistle,
+      Hardfork.SpuriousDragon,
+      Hardfork.Byzantium,
+      Hardfork.Constantinople,
+      Hardfork.Petersburg,
+      Hardfork.Istanbul,
+      Hardfork.MuirGlacier,
+      Hardfork.Berlin,
+      Hardfork.London,
+      Hardfork.ArrowGlacier,
+    ]
+    if (!supportedHardforks.includes(this._common.hardfork() as Hardfork)) {
+      throw new Error(
+        `Hardfork ${this._common.hardfork()} not set as supported in supportedHardforks`
```

Comment:
Small nit, won't make this a blocker though: I guess you can write error messages "one level higher", so end users do not really have a notion of this internal "supportedHardforks" variable and the like, so something like "The provided hardfork {hardfork} is not supported by the VM" is likely better readable/less confusing.

If someone thinks about updating this along some other PR would be nice, but totally no blocker as well.

Reply:
Yeah I think that makes sense. I was thinking of trying to tackle the London-as-default-hardfork issue, so I could add it that PR